openquake/hazardlib/gsim/kotha_2019.py

```diff
@@ -0,0 +1,375 @@
+    5.000  1.766244279359640  2.775644704909250   0.136811181908039  1.277024063081970  -1.134940922975580   3.156427441541180  4.182103070858070   4.099969082316580  0.247128194747863  -0.001550196672696  0.625986200151889  0.000000000000000  0.598171990718161  0.357769751705229  0.250782161006834  0.409277420621721  3.27278257182859  -0.529012351103819  0.397002875689376  3.84718134870989  -0.601713782510972  0.476960884877633
+    6.000  1.495644634092600  2.906956649834780   0.206605866198126  1.436150033009390  -1.155226946853440   2.732111686024190  5.157196005146210   5.134139700462980  0.261450980486218  -0.001439766119884  1.343568797412350  0.000000000000000  0.577497461758988  0.364322510724063  0.224152579183778  0.401597199934827  3.01673206668580  -0.489064156495419  0.379679274738408  3.58229829260865  -0.561049974091566  0.459592859585885
+    7.000  1.182821190754860  2.917667033954530   0.223647052589440  1.645537964856860  -1.228574871054330   2.970825808343180  5.783138212668300   5.347061845422780  0.289575079930824  -0.000867891160389  1.439390706721320  0.000000000000000  0.563819357916120  0.368571233598105  0.229328741533368  0.401052676368561  3.09654599365856  -0.503275832203786  0.363962303516824  3.44766788599524  -0.540181500989622  0.447668743764740
+    8.000  0.877936104327193  2.908933205097090   0.234947578468289  1.691937861016660  -1.271209905469500   3.052069542435400  6.188468047371990   6.198930745809970  0.312385151596051  -0.000600200472803  1.671791342658880  0.000892670164182  0.556359572194014  0.365797914105721  0.204945101304949  0.401323166589614  3.08002142877393  -0.501377187109764  0.359647250033555  3.38667681338371  -0.530743607500501  0.440968433335620
+    """)
+
+    CONSTANTS = {"Mref": 4.5, "Rref": 30., "Mh": 6.2}
+
+
+## Adjustment factors based on the "value" term of a three-point discrete
+## approximation of a normal distribution according to Miller & Rice (1983)
+#CFACT = 1.732051
+#
+#
+## For the regional coefficients a "faster" attenuation means the c3 term
+## is made more strongly negative, and vice-versa for a "slower" attenuation
+#REGION_COEFFS = {"Average": 0., "Faster": -CFACT, "Slower": CFACT}
+#
+#
+## For other distributions (stress factor and sigma_mu) the coefficients are
+## increased or decreased by the "value" term
+#OTHER_COEFFS = {"Low": -CFACT, "Middle": 0., "High": CFACT}
+
+
+class KothaEtAl2019SERA(KothaEtAl2019):
+    """
+    Implementation of the Kotha et al. (2019) GMPE with the site
+    amplification components included. This form of the GMPE defines the
+    site in terms of a measured or inferred Vs30, with the total
+    aleatory variability adjusted accordingly.
+    """
+
+    #: Required site parameter is not set
+    REQUIRES_SITES_PARAMETERS = set(("vs30", "vs30measured"))
```

Comment:
Off the topic. I know this is also used in other GMPEs. However, thinking at this now. Is it correct to change aleatory variability on the basis of an epistemic problem? Shouldn't aleatory variability be irreducible?

Reply:
On a theoretical level, yes, and I would argue that for site specific analysis then uncertainty on the site property itself this should be treated as an epistemic uncertainty. Though for a site specific analysis you would not be following this approach anyway. On a practical level, it is inevitable that for regional scale risk analysis you will have (sometimes considerable) uncertainties on the site parameter, but it would be extremely difficult to treat these epistemically in the way that OpenQuake (or any other software) currently does. So unless you accept a higher sigma in the ground motion variability it is highly likely that this uncertainty will just get dropped out of the hazard and risk model altogether, and I would argue that is worse than treating epistemic as aleatory for the time being. If you have been running large scale hazard and risk calculations using Vs30 from Wald & Allen but applying a GMPE calibrated whose site term is calibrated on a measured Vs30, then so far this site parameter uncertainty has been omitted (or otherwise substantially underestimated) from the analysis.